packages/slate-react/src/components/editable.tsx

```diff
@@ -289,11 +289,11 @@ export const Editable = (props: EditableProps) => {
 
         const anchorNodeSelectable =
           hasEditableTarget(editor, anchorNode) ||
-          isTargetInsideVoid(editor, anchorNode)
+          isTargetInsideNonReadonlyVoid(editor, anchorNode)
```

Comment:
@ahoisl 👋🏽 I ran into this issue today. I'm wondering if the `||` needs to be `&&` instead.

Reply:
Are you sure it's the same issue? "Cannot resolve from DOM point" is a very common slate error.

I'm pretty sure the boolean operation (`||`) is correct in this case. "void" nodes can usually be selected, just not when they are readonly and that's the only thing this fix is really checking for.

But there might be other issues or use cases that probably need to be considered?

Reply:
I actually think the issue is with the implementation of `isTargetInsideNonReadonlyVoid`. 

https://github.com/ahoisl/slate/blob/43f84e489f2a5012acb3a7c3e296ad7d4a645ce8/packages/slate-react/src/components/editable.tsx#L1414..1422. 

It's actually implemented as "!isReadOnly || isVoid" instead of "!isReadOnly && isVoid". The double negative is confusing. I think that's where the confusion comes from. I think [this](https://github.com/ianstormtaylor/slate/pull/4887) fixes it

Reply:
The fix in this PR #4727 was really just intended to avoid the "Cannot resolve from DOM point" error in readonly void nodes. I tried to change code as minimal as possible. The `onDOMSelectionChange` already had the `|| isTargetInsideVoid` in there, I just made sure it's not doing anything on readonly void nodes.

Your PR #4887 now changes the behavior to not step into the if-clause and call `toSlateRange` for non-readonly void nodes. I'm not sure if this is correct/intended or not.